core/modules/layout/layout.admin.inc

```diff
@@ -1129,6 +1130,133 @@ function layout_block_configure_redirect($layout, $block_uuid) {
 }
 
 /**
+ * Menu callback; Remove a block from the front end contextual link.
+ */
+function layout_block_contextual_remove_form($form, &$form_state, $layout, $block_uuid) {
+  backdrop_set_title(t('Remove block'));
+  $delta = $layout->content[$block_uuid]->delta;
+  $module = $layout->content[$block_uuid]->module;
+  $block_info = layout_get_block_info($module, $delta);
+  $block_title = $form_state['block_title'] = $block_info['info'];
+
+  // Check if this block is placed in multiple layouts. Custom and hero blocks
+  // are only available to one layout, so no need to check for multiple layouts.
+  if ($delta != 'custom_block' && $delta != 'hero') {
+    $usage = layout_get_block_usage($module, $delta);
+    $layouts = $form_state['layouts'] = layout_load_multiple(array_keys($usage));
+
+    // Exclude admin default
+    if (isset($layouts['admin_default'])) {
+      unset($layouts['admin_default']);
+    }
+
+    $options = array();
+    if (count($layouts) > 1) {
+      foreach ($usage as $layout_name => $active_layout) {
+        foreach ($active_layout as $region_name => $block_instance) {
+          $multiple_blocks = (count($block_instance) > 1);
+          foreach ($block_instance as $index => $block) {
+            $current_layout = $layouts[$layout_name];
+            $path = $current_layout->getPath();
+            $path = !empty($path) ? ' (' . $path . ')' : t(' (all paths using default layout)');
+            $region_label = t(' in ') . check_plain($region_name) . t(' region');
+            $block_index = '';
+            if ($multiple_blocks) {
+              $block_index = t(' block ') . ($index + 1);
+            }
+            $options[$block->uuid] = $current_layout->title . $path . $region_label . $block_index;
+          }
+        }
+      }
+
+      $form['remove_blocks'] = array(
+        '#type' => 'checkboxes',
+        '#title' => t('Remove block %block_title from the following layouts (paths)', array('%block_title' => $block_title)),
+        '#options' => $options,
+        '#default_value' => array($block_uuid),
+      );
+    }
+  }
+
+  if (!isset($form['remove_blocks'])) {
+    $form['message'] = array(
+      '#markup' => t('Remove this block?'),
+    );
+    $form_state['layout'] = $layout;
+    $form_state['block_uuid'] = $block_uuid;
+  }
+
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Remove'),
+  );
+  $form['actions']['cancel'] = array(
+    '#type' => 'submit',
+    '#value' => t('Cancel'),
+    '#attributes' => array(
+      'class' => array('button-secondary',),
+    ),
```

Comment:
I think other places we have Cancel links within dialogs we use `#type = link`, e.g.

```
  $options['attributes']['class'][] = 'form-cancel';
  $form['actions']['cancel'] = array(
    '#type' => 'link',
    '#title' => t('Cancel'),
    '#href' => $options['path'],
    '#options' => $options,
  );
```

I think that is supported within dialogs, the link then appears as a button.